pkg/vm/interop.go

```diff
@@ -126,12 +126,25 @@ func init() {
 
 // EnumeratorCreate handles syscall System.Enumerator.Create.
 func EnumeratorCreate(v *VM) error {
-	data := v.Estack().Pop().Array()
-	v.Estack().Push(&Element{
-		value: stackitem.NewInterop(&arrayWrapper{
+	var interop interface{}
+	switch t := v.Estack().Pop().value.(type) {
+	case *stackitem.Array, *stackitem.Struct:
+		interop = &arrayWrapper{
+			index: -1,
+			value: t.Value().([]stackitem.Item),
+		}
+	default:
+		data, err := t.TryBytes()
```

Comment:
1) 2cd68091071c5e3d9d08f640568f462c2b3a66ed is more about #1201 than #1055.
2) So we would allow any primitive type here? I don't think that's what is intended, should it work for integers?

Reply:
1. Updated PR description, c9ef742 and 459ac34 close #1201.
2. We should support all primitive types here (these are [Integer](https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/Types/Integer.cs#L9), [Boolean](https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/Types/Boolean.cs#L9), [ByteArray](https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/Types/ByteString.cs#L9)) and Buffer (which is [not primitive](https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/Types/Buffer.cs#L9) in C#, but has [GetSpan method](https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/Types/Buffer.cs#L57)) because [click](https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Iterator.cs#L24) and [click](https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Iterator.cs#L23). In Go *all* of these 4 types and *only* these 4 types do not return an error after `TryBytes()` call, so I decide to use this capability in `default` case. Iterating over integers means that we should convert integer to a byte slice and then iterate over it (although it's a bit strange).